common/persistence/client/quotas.go

```diff
@@ -32,26 +32,45 @@ import (
 var (
 	CallerTypePriority = map[string]int{
 		headers.CallerTypeAPI:        0,
-		headers.CallerTypeBackground: 1,
+		headers.CallerTypeBackground: 2,
 	}
 
 	APIPriorityOverride = map[string]int{
 		"GetOrCreateShard": 0,
 		"UpdateShard":      0,
 
-		// this is a preprequisite for checkpoint queue process progress
+		// This is a preprequisite for checkpoint queue process progress
 		"RangeCompleteHistoryTasks": 0,
+
+		// Task resource isolation assumes task can always be loaded.
+		// When one namespace has high load, all task processing goroutines
+		// may be busy and consumes all persistence request tokens, preventing
+		// tasks for other namespaces to be loaded. So give task loading a higher
+		// priority than other background requests.
+		// NOTE: we also don't want task loading to consume all persistence request tokens,
+		// and blocks all other operations. This is done by limiting the total rps allow
+		// for this priority.
+		// TODO: exclude certain task type from this override, like replication.
+		"GetHistoryTasks": 1,
 	}
 
-	RequestPrioritiesOrdered = []int{0, 1}
+	RequestPrioritiesOrdered = []int{0, 1, 2}
+
+	PriorityRatePercentage = map[int]float64{
+		0: 1.0,
+		1: 0.8,
```

Comment:
I am not sure about this actually. 

When # of task processing goroutines is too high and causing persistence busy errors, all namespace will be affected I think (not because of task loading, just the processing itself)? To solve this problem, we can 
1. Reduce # of goroutines until there's no persistency service busy
2. Specify per ns persistence request quota so that task processing will only fail for high throughput namespaces. 

In both cases, task loading requests should be able to go through and we won't have the loading issue? Well I guess it also depends on how we allocate the per ns persistence quota.

Reply:
Is the intention of 0.8 for priority 1 to avoid task loading use up all available quota? 
If task loading consume too much quota, it means task processing is staving so in-memory task will pile up and have backlog. Can we slow down task loading if there is enough task buffered?

Reply:
Yes. That's the intention. 

We already have the logic to stop task loading when a shard's task buffer is full. But the behavior is binary, before the buffer is full, it will load task as fast as possible and then stop loading if buffer is full. Maybe make the behavior linear?

But from multi-cursor design perspective, task loading should always be as fast as possible, and if buffer is full, we unload tasks for some namespaces and keep loading new tasks as we can't know if the next task in DB should be immediately processed or throttled before loading.

So I kinda think allow task loading to a certain percentage of the total db resource makes sense. 

Another idea is use the per namespace persistence ratelimiting. I need to think more on this.